lib/datadog/statsd.rb

```diff
@@ -84,6 +84,7 @@ def initialize(
       buffer_overflowing_stategy: :drop,
 
       logger: nil,
+      auto_close: false,
```

Comment:
Why not having it set to true? I think it would be a better default behavior?
It is important that two calls to `close` in a row are idempotent anyway.

Reply:
I just don't want to change the default behavior of the library too much. `at_exit` is used for example by sinatra to run the applications after all the files are read. If we change this to true, it would mean any existing sinatra application monitored with the library would stop reporting metrics.